src/engine/BMGame.php

```diff
@@ -476,10 +380,13 @@ protected function do_next_step_choose_join_game() {
     }
 
     protected function update_game_state_choose_join_game() {
-        if (isset($this->hasPlayerAcceptedGameArray) &&
-            is_array($this->hasPlayerAcceptedGameArray) &&
-            in_array(FALSE, $this->hasPlayerAcceptedGameArray)) {
-            foreach ($this->hasPlayerAcceptedGameArray as $playerIdx => $hasAccepted) {
+        $hasPlayerAcceptedGameArray = array_fill(0, $this->nPlayers, NULL);
+        foreach ($this->playerArray as $playerIdx => $player) {
+            $hasPlayerAcceptedGameArray[$playerIdx] = $player->hasPlayerAcceptedGame;
+        }
+
+        if (in_array(FALSE, $hasPlayerAcceptedGameArray)) {
+            foreach ($hasPlayerAcceptedGameArray as $playerIdx => $hasAccepted) {
```

Comment:
So, one argument i could make here is: in the new refactored world, any time you're accessing `$playerIdx` inside BMGame, it's worth thinking about why, because why can't you just act on each `$player` object without having to futz with indices?

And in this case, i think that question is legit --- in this case, the only reason you're pulling `$this->playerArray` with indices is so you can construct your own temporary array.  But i'm not sure you need that --- IMO it would be tidier to separate the "find out if any player has not accepted" part from the "set each player's waitingOnAction" part.  Because the latter part you can do by setting:
```php
$player->waitingOnAction = !$player->hasPlayerAcceptedGame
```

There's no need to reference a BMGame temporary variable for that.